build.yaml

```diff
@@ -31,14 +26,34 @@ steps:
        valueFrom: default_ns.name
    public: false
    secrets:
-    # hail|test-batch
-    - user-jwt-vkqfw
-    - gsa-key-2x975
-    # pipeline|test-user
-    - pipeline-test-0-1--hail-is-service-account-key
+    - test-gsa-key
    dependsOn:
     - default_ns
     - deploy_batch_sa
+ - kind: runImage
+   name: create_accounts
+   image: "{{ base_image.image }}"
+   script: |
+     set -e
+     echo creating jwt-signing-key secret...
+     python3 -c "
+     import secrets, json, jwt
+     secret_key = secrets.token_bytes(64)
+     with open('secret-key', 'wb') as f:
+       f.write(secret_key)
+     payload = {'ksa_name':'unused','gsa_email':'hail-test@hail-vdc.iam.gserviceaccount.com','bucket_name':'hail-test','gsa_key_secret_name':'test-gsa-key'}
+     with open('jwt', 'w') as f:
+       f.write(jwt.encode(json.dumps(payload), secret_key))
+     "
+     kubectl -n {{ default_ns.name }} create secret generic jwt-secret-key --from-file=./secret-key
+     kubectl -n {{ default_ns.name }} create secret generic batch-jwt --from-file=./jwt
+     kubectl -n {{ default_ns.name }} create secret generic ci2-jwt --from-file=./jwt
+     kubectl -n {{ batch_pods_ns.name }} create secret generic test-jwt --from-file=./jwt
+   serviceAccount: ci2-agent
```

Comment:
Why is this necessary? Where does the gsa_key come from for this service account?

Reply:
This is not a Hail service account, it is a k8s service account.  It has no gsa key.  It has the permissions necessary to run the kubectl create secret commands above in both the default and batch_pods namespaces.